### PR TITLE
@trezor/blockchain-link 2.0.0

### LINKED ISSUE #3326: [blockchain-link]: Refactor workers to modules
Refactor worker files to use them as a modules

### PR DESCRIPTION
**Pull request by szymonlesisz**
fix https://github.com/trezor/trezor-suite/issues/3326
prerequisite for https://github.com/trezor/trezor-suite/pull/4704 and https://github.com/trezor/trezor-suite/pull/4500 
### Unify typescript and webpack settings with the rest of monorepo packages:
- using stricter typescript compilerOptions
- using `import type ...` syntax
- reducing polyfills from typescript transpilation in ./lib. packed/unpacked size: 32.9/222.6 kB > 27.5/151.9 kB
- using `ts-loader` instead of `babel-loader`
- removed @babel/runtime and multiple unused dependencies

### Refactored architecture of workers.
They may now be used as commonjs module in main context: 
```
import BlockchainLink from '@trezor/blockchain-link';
import Blockbook from '@trezor/blockchain-link/lib/workers/blockbook';

const link = new BlockchainLink({
    name: 'bitcoin',
    worker: Blockbook,
    server: ['url1.of.the.be', 'url2.of.the.be'],
    debug: true,
});
```
or in WebWorker context like before:
```
import BlockchainLink from '@trezor/blockchain-link';
import BlockbookWorker from 'worker-loader?filename=workers/blockbook-worker.[hash].js!@trezor/blockchain-link/lib/workers/blockbook/index.js';

const link = new BlockchainLink({
    name: 'bitcoin.',
    worker: BlockbookWorker,
    server: ['url1.of.the.be', 'url2.of.the.be'],
    debug: true,
});
```

### Removed `build` directory from npm registry. 
Unnecessary overhead in npm.
Implementation (trezor-connect) should build them using own webpack config (in fact it already does)

### CODE REVIEW

File: packages/blockchain-link/src/utils/ws.ts
Comment (marekrjpolak) [inline packages/blockchain-link/src/utils/ws.ts:4]: It probably isn't related to this PR, but is it necessary to have both `ws.ts` and `ws.js` next to each other?

**Review by marekrjpolak — APPROVED** [review]
Looks good and seems to be working. Moreover, I've managed to quickly create a dirty fix so it works with Electrum worker as well. But of course I'll have to do it properly after merging.

File: packages/blockchain-link/README.md
Comment (karliatto) [inline packages/blockchain-link/README.md:66]: What about `yarn workspace @trezor/blockchain-link test:integration` shouldn't we include also the `integration` tests?

File: packages/blockchain-link/src/utils/ws.ts
Comment (szymonlesisz) [reply packages/blockchain-link/src/utils/ws.ts:4]: totally right, removed not used file

File: packages/blockchain-link/README.md
Comment (szymonlesisz) [reply packages/blockchain-link/README.md:66]: test:integration section added

File: packages/blockchain-link/package.json
Comment (Nodonisko) [inline packages/blockchain-link/package.json:53]: Could you please try to upgrade `ws` to version 7.5.5 or higher? We currently have like 5 different version of `ws` in project and I want to remove at least some of them. If this will cause problems then we can leave it as it is, but 5 deferent versions is quite a lot.

Screenshot 2022-01-18 at 12 33 53

Also related to #4760

Comment (szymonlesisz) [reply packages/blockchain-link/package.json:53]: i think it should be done in separate PR, `ws` needs to be updated across the whole monorepo packages not only blockchain-link and this PR is about blockchain-link modules :) i already ended up with much more code changes than i wanted

Comment (Nodonisko) [reply packages/blockchain-link/package.json:53]: thanks for explanation, resolving

**Comment by szymonlesisz** [comment]
last minute change: package version fixup 2.0.0-beta > 2.0.0

**Review by marekrjpolak — APPROVED** [review]
Still like it. :shipit:

**Review by mroz22 — APPROVED** [review]
looks good. once this is in develop I am going to enabled blockchain-link integration test for schedules pipelines. not really needed to block this PR.

**Pull request merged**
